src/solver/impls/petsc/petsc.cxx

```diff
@@ -669,15 +669,22 @@ PetscErrorCode solver_rhsjacobian(TS UNUSED(ts), BoutReal UNUSED(t), Vec UNUSED(
   PetscFunctionReturn(0);
 }
 #else
-PetscErrorCode solver_rhsjacobian(TS ts,BoutReal t,Vec globalin,Mat *J,Mat *Jpre,MatStructure *str,void *f_data) {
+PetscErrorCode solver_rhsjacobian(MAYBE_UNUSED(TS ts), MAYBE_UNUSED(BoutReal t),
+                                  MAYBE_UNUSED(Vec globalin), Mat *J, Mat *Jpre,
+                                  MAYBE_UNUSED(MatStructure *str),
+                                  MAYBE_UNUSED(void *f_data)) {
   PetscErrorCode ierr;
 
   PetscFunctionBegin;
-  ierr = MatAssemblyBegin(*Jpre, MAT_FINAL_ASSEMBLY);CHKERRQ(ierr);
-  ierr = MatAssemblyEnd(*Jpre, MAT_FINAL_ASSEMBLY);CHKERRQ(ierr);
-  if (*J != *Jpre){
-    ierr = MatAssemblyBegin(*J, MAT_FINAL_ASSEMBLY);CHKERRQ(ierr);
-    ierr = MatAssemblyEnd(*J, MAT_FINAL_ASSEMBLY);CHKERRQ(ierr);
+  ierr = MatAssemblyBegin(*Jpre, MAT_FINAL_ASSEMBLY);
+  CHKERRQ(ierr);
```

Comment:
This is probably automatic formatting, but having `CHKERRQ` on the same line is the pattern usually used in PETSc code.

Reply:
Ah, good point. Might need to find a way to tell clang-format to leave PETSc stuff alone